### ISSUE TITLE
Incoming iOS calls not connected using STUN, when launched from VoIP push and app hasn't been running

### ISSUE THREAD

**Issue by dimonka** [matrix-org/matrix-ios-sdk#1326]
**Describe the bug**
We're developing a messenger app and faced an issue with call connection.

If the app is deliberately killed (swiped up from the app list), or the phone has been idle for a long while, when the call is received and answered, connection does not happen and remains in "Connecting" state.

It does not depend if the screen was locked or not. If I terminate the app, and make a call to that device, VoIP is received, it starts my app, but still the call stucks in "Connecting" state. If I hang up the call, and call again - everything is fine. I can lock the screen or switch to another app - still it's gonna be OK. But the first call after wakeup from VoIP push - NO. Stucks in connecting...

Oh, I forgot to say - that only happens when devices are in different networks, and Turn/Stun server is used. We're using standard Coturn-4.5.0.7 'dan Eider'...

MatrixSDK is 0.17.4

Also, MatrixSDK.answer() method executes successfully (tested in debugger), but after that I have no idea where to dig. I dont know how to debug possible Stun issues... In fact, does Stun really care how my app was launched?... no.

Any ideas?

**To Reproduce**
1. Kill my app on iOS device. Swipe it up from app list - that terminates the app
2. Make a call to that device. I use Android version of Element, but that does not depend
3. Answer the call on iOS device
4. If the screen was locked, the call timer will start on iOS device, with native CallKit UI (probably because .answer() was successful and .fulfil() was called too)
5. If the screen wasnt locked, my app UI will appear, the call UI (our personal) will appear too, and the process will hang waiting for "Connected" state
6. Call state remains "Connecting", on both devices
7. P.S. If I make another call right after the first one - it's gonna be all fine and will connect (because app is already started by the first call)

**Expected behavior**
A call is connected

**Smartphone (please complete the following information):**
 - Device: tested on iPhone X and iPhone 12
 - OS: iOS 15
 - Matrix SDK 0.17.4
 - Coturn 4.5.0.7
 - Synapse 1.47.0rc2

**Comment by dimonka** [comment]
There is one clue. For each call I usually receive 6-7 "m.call.candidate" events. Most of them are some garbage with local IPs, but one is the ONE :) with proper caller IP.

Now, when app is started from VoIP push, I don't receive "m.call.candidate" event with proper caller IP. Only garbage ones. I see the proper event in the database later, but do not receive it in my app. When call happens in a started app - event is received.

How can it be?

**Comment by dimonka** [comment]
Well, If anyone interested - I've solved id. But it really seem to be an SDK bug. The proper call candidate info does not arrive.

I had to make a separate block of code after accepting it, which waits for 2 seconds (for instance), and then checks - if the cal status is still .connecting - retrieve last candidate events from room where the call was made, and process them again. It worked.

Here's an example. It's a code from `func provider(_ provider: CXProvider, perform action: CXAnswerCallAction)`

![image](https://user-images.githubusercontent.com/15159834/146410056-ced4495f-5a81-4732-bf4e-95bd8d58db1a.png)
